xds/internal/balancer/clusterimpl/clusterimpl.go

```diff
@@ -327,14 +341,16 @@ func (b *clusterImplBalancer) ExitIdle() {
 func (b *clusterImplBalancer) UpdateState(state balancer.State) {
 	b.serializer.TrySchedule(func(context.Context) {
 		b.childState = state
-		b.ClientConn.UpdateState(balancer.State{
-			ConnectivityState: b.childState.ConnectivityState,
-			Picker: b.newPicker(&dropConfigs{
-				drops:           b.drops,
-				requestCounter:  b.requestCounter,
-				requestCountMax: b.requestCountMax,
-			}),
-		})
+		if !b.inhibitPickerUpdates {
```

Comment:
nit: Can invert this condition to return early, reducing one level of indentation.
```go
if b.inhibitPickerUpdates {
    return
}
// update picker.
```